### ISSUE TITLE
Can't close actionSheet

### ISSUE THREAD

**Issue by DedicatedDev** [Daniel-Ioannou/flutter_adaptive_action_sheet#22]
ActionSheet is not closed. 
I've implemented Navigator.of(context).pop()
it close parent widget not actionsheet.

**Comment by Daniel-Ioannou** [comment]
@DedicatedDev Not sure what's the problem. Can you please provide a small example?

**Comment by DedicatedDev** [comment]
Thank you. I've tried to implement closed solution but it is not working. 
I'm developing quiz app right now. Just replaced this role native module -> CupertinoActionSheetAction
implementation code.  disabled part is for current this repo. 
Navigator.of(context).pop don't close actionsheet. 

showCupertinoModalPopup(
      context: context,
      builder: (BuildContext context) => CupertinoActionSheet(
        title: const Text('Edit Photo'),
        actions: [
          CupertinoActionSheetAction(
            child: const Text('Take New Photo',
                style: TextStyle(color: Colors.black)),
            onPressed: () {
              Navigator.pop(context);
              getImagFromGallery(ImageSource.camera);
            },
          ),
          CupertinoActionSheetAction(
            child: const Text(
              'Upload New Photo',
              style: TextStyle(color: Colors.black),
            ),
            onPressed: () {
              Navigator.pop(context);
              getImagFromGallery(ImageSource.gallery);
            },
          ),
          CupertinoActionSheetAction(
            child: const Text('Delete', style: TextStyle(color: Colors.black)),
            onPressed: () {
              Navigator.pop(context);
            },
          ),
          CupertinoActionSheetAction(
            child: const Text('Cancel'),
            onPressed: () {
              Navigator.pop(context);
            },
          ),
        ],
      ),
    );
    // showAdaptiveActionSheet(
    //   context: context,
    //   title: const Text('Edit Photo'),
    //   actions: <BottomSheetAction>[
    //     BottomSheetAction(
    //         title: const Text('Take New Photo'),
    //         onPressed: () => Navigator.pop(context)),
    //     BottomSheetAction(
    //         title: const Text('Upload New Photo '),
    //         onPressed: () {
    //           //Navigator.of(context).pop();
    //         }),
    //     BottomSheetAction(title: const Text('Delete'), onPressed: () {}),
    //   ],
    //   cancelAction: CancelAction(
    //       title: const Text(
    //           'Cancel')), // onPressed parameter is optional by default will dismiss the ActionSheet
    // );

![image](https://user-images.githubusercontent.com/62229912/115923935-b2120e00-a487-11eb-9ffe-9d70ebaf3e66.png)

**Comment by DedicatedDev** [comment]
![image](https://user-images.githubusercontent.com/62229912/115923996-c950fb80-a487-11eb-8953-6eb764a4c6a7.png)

**Comment by DedicatedDev** [comment]
I've implemented Navigator.of(context).pop().
If I click one time, it does not closed and navigate parent widget. 
so screen move to previous screen. 
![image](https://user-images.githubusercontent.com/62229912/115924313-3a90ae80-a488-11eb-89fd-fdeecbf1e54a.png)

**Comment by Daniel-Ioannou** [comment]
@DedicatedDev That weird behavior, I try to reproduce it but I couldn't. I copy paste the code you provided to a example project and works as expected. If you check the lib code you will see that basically it calls `showCupertinoModalPopup` for ios and `showModalBottomSheet` for Android.  `Navigator.of(context).pop()` should pop the top-most route off the navigator, in this case is the ActionSheet and not the parent widget.

Please check the code that I used to reproduce the issue and and let me know if I missed anything
  ```Dart
    showAdaptiveActionSheet(
      context: context,
      title: const Text('Edit Photo'),
      actions: [
        BottomSheetAction(
          title: const Text('Take New Photo'),
          onPressed: () => Navigator.pop(context),
        ),
        BottomSheetAction(
          title: const Text('Upload New Photo '),
          onPressed: () {
            Navigator.of(context).pop();
          },
        ),
        BottomSheetAction(title: const Text('Delete'), onPressed: () {}),
      ],
      cancelAction: CancelAction(
        title: const Text('Cancel'),
      ), 
    );
  ```

**Comment by naoki0719** [comment]
I think you're trying to pop a nested Navigator.
`Navigator.of(context, rootNavigator: true).pop();` should fix it.

**Comment by DharmbirChoudhary** [comment]
@naoki0719 Yes it works :)

**Comment by ricardogobbo** [comment]
I am facing the exact same error, but just on iOS platform.

**Comment by KCL-TelferAA** [comment]
> Navigator.of(context, rootNavigator: true).pop();



> Normally the mechanic of an action sheet to to automatically remove its self after any sections. and so we should not have to call pop to remove the action sheet.

I agree with this. I thought it strange that after clicking an option, the sheet did not close.
